File: problems/14.longest_common_prefix.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) 
    {
        if (strs.Length == 0) {
            return "";
        }
        string common_prefix = strs[0];
        char[] common_prefix_arr = common_prefix.ToCharArray();
        int common_prefix_len = common_prefix.Length;

        foreach(string str in strs) {
            if (common_prefix_len == 0) {
                break;
            }
            char[] str_arr = str.ToCharArray();
            common_prefix_len = common_prefix_len > str_arr.Length ? str_arr.Length : common_prefix_len;
            for (int i = 0; i < common_prefix_len; i++) {
                if (str_arr[i] != common_prefix[i]) {
                    common_prefix_len = i;
                    continue;
                }
            }
        }

        string result = "";
        for (int i = 0; i < common_prefix_len; i++) {
            result += common_prefix[i];
        }

        return result;
    }
}

File: interview_top/14.longest_common_prefix.go
package interviewtop

func longestCommonPrefix(strs []string) string {
	r := len(strs[0])
	for i := 1; i < len(strs) && r > 0; i++ {
		if len(strs[i]) < r {
			r = len(strs[i])
		}
		for j := 0; j < r; j++ {
			if strs[i][j] != strs[0][j] {
				r = j
				break
			}
		}
	}
	return strs[0][:r]
}
